File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-4.java
class Solution {
    public boolean mergeTriplets(int[][] triplets, int[] target) {
        int x = 0, y = 0, z = 0;
        for (int[] trip : triplets) {
            if (trip[0] > target[0] || trip[1] > target[1] || trip[2] > target[0]){
                continue;
            }
            System.out.println("ok");
            if (trip[0] == target[0])
                x = 1;
            if (trip[1] == target[1])
                y = 1;
            if (trip[2] == target[2])
                z = 1;
        }
        System.out.println(x + " " + y + " " + z);
        return x == 1 && y == 1 && z == 1;
    }
}

File: Data Structures & Algorithms/4sum/submission-4.cpp
class Solution {
    std::vector<std::vector<int>> solution;
    std::vector<int> current;
public:
    vector<vector<int>> fourSum(vector<int>& nums, int target) {
        std::sort(nums.begin(), nums.end());
        ksum(nums, target, 4, 0);
        return solution;
    }
    void ksum(vector<int>& nums, int target, int k, int start) {
        if (k == 2) {
            int left = start;
            int right = nums.size() - 1;
            while (left < right) {
                int a{nums[left]}, b{nums[right]};
                if (a + b == target) {
                    current.push_back(a);
                    current.push_back(b);
                    solution.push_back(current);
                    current.pop_back();
                    current.pop_back();
                    left ++;
                    right --;
                    while (left != nums.size() && nums[left] != a && left < right) left ++;
                    while (nums[right] != b && left < right) right --;
                }
                else if (a + b > target){
                    left ++;
                    while (left != nums.size() && nums[left] != a && left < right) left ++;
                }
                else {
                    right --;
                    while (nums[right] != b && left < right) right --;
                }
            }
        }
        else {
            for (int i{start}; i < nums.size(); ++i) {
                current.push_back(nums[start]);
                ksum(nums, target - nums[start], k - 1, i + 1);
                current.pop_back();
            } 
        }
    }
};